pkg/doc/signature/verifier/verifier.go

```diff
@@ -83,19 +67,19 @@ func New(resolver keyResolver, suites ...SignatureSuite) (*DocumentVerifier, err
 }
 
 // Verify will verify document proofs
-func (dv *DocumentVerifier) Verify(jsonLdDoc []byte) error {
+func (dv *DocumentVerifier) Verify(jsonLdDoc []byte, opts ...jsonld.ProcessorOpts) error {
 	var jsonLdObject map[string]interface{}
 
 	err := json.Unmarshal(jsonLdDoc, &jsonLdObject)
 	if err != nil {
 		return fmt.Errorf("failed to unmarshal json ld document: %w", err)
 	}
 
-	return dv.verifyObject(jsonLdObject)
+	return dv.verifyObject(jsonLdObject, append(opts, jsonld.WithRemoveAllInvalidRDF()))
```

Comment:
May be I am missing something here, shouldn't we keeping invalid RDFs as it is so that verifier should fail when it encounters one.

-> signer should remove invalid RDFs before signing
-> verifier should verify as it is.

Reply:
Yes - that is what are trying to do.

Reply:
I will re-check it in a few hours, come back with the answer. An important question!

Reply:
I changed that, now `jsonld.WithRemoveAllInvalidRDF()` is applied on demand. To facilitate VC, I added `WithJSONLDOnlyValidRDF()` option.